pkg/types/config/interfaces.go

```diff
@@ -0,0 +1,81 @@
+package config
+
+import (
+	corecontrollers "github.com/rancher/wrangler/v3/pkg/generated/controllers/core"
+	core "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+	wrbac "github.com/rancher/wrangler/v3/pkg/generated/controllers/rbac/v1"
+	"github.com/rancher/wrangler/v3/pkg/generic"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+)
+
+// nonNamespacedCacheAndClient limits a Controller interface to only the direct and cached clients
+type nonNamespacedCacheAndClient[T generic.RuntimeMetaObject, TList runtime.Object] interface {
```

Comment:
This type is not used anywhere, but I guess it will be needed in the future, right?